cli/tauri-bundler/src/bundle/osx_bundle.rs

```diff
@@ -87,9 +89,215 @@ pub fn bundle_project(settings: &Settings) -> crate::Result<Vec<PathBuf>> {
     create_bootstrapper(&bundle_directory, settings)
       .with_context(|| "Failed to create OSX bootstrapper")?;
   }
+
+  if let Some(identity) = settings.osx_signing_identity() {
+    sign(app_bundle_path.clone(), identity, &settings)?;
+    match notarize_auth_args() {
+      Ok(args) => {
+        notarize(app_bundle_path.clone(), args, settings)?;
+      }
+      Err(e) => {
+        common::print_info(format!("skipping app notarize, {}", e.to_string()).as_str())?;
+      }
+    }
+  }
+
   Ok(vec![app_bundle_path])
 }
 
+fn sign(app_bundle_path: PathBuf, identity: &str, settings: &Settings) -> crate::Result<()> {
```

Comment:
I think it would be a good idea to move all this logic into a separate file as this one is quite long and complicated now.

Reply:
yeah I agree with you, we should do it when this is properly working :D

Reply:
I think i'll restructure the bundler crate on a separate PR.